Gordon360/Services/HousingService.cs

```diff
@@ -251,77 +251,77 @@ public int EditApplication(string editorId, string sess_cde, int apartAppId, str
-                if (applicantResult == null)
-                {
-                    throw new ResourceNotFoundException() { ExceptionMessage = "Applicant with ID " + id + " could not be inserted." };
-                }
-            }
-
-            foreach (string id in applicantIdsToRemove)
-            {
-                IEnumerable<ApartmentApplicantViewModel> applicantResult = null;
-
-                // All SqlParameters must be remade before being reused in an SQL Query to prevent errors
-                //idParam.Value = id; might need if this ODD solution is not satisfactory
-                appIdParam = new SqlParameter("@APPLICATION_ID", apartAppId);
-                idParam = new SqlParameter("@ID_NUM", id);
-                sessionParam = new SqlParameter("@SESS_CDE", sess_cde);
-
-                applicantResult = RawSqlQuery<ApartmentApplicantViewModel>.query("DELETE_AA_APPLICANT @APPLICATION_ID, @ID_NUM, @SESS_CDE", appIdParam, idParam, sessionParam); //run stored procedure
-                if (applicantResult == null)
-                {
-                    throw new ResourceNotFoundException() { ExceptionMessage = "Applicant with ID " + id + " could not be removed." };
-                }
-            }
+            //IEnumerable<ApartmentChoiceSaveViewModel> apartmentChoicesResult = null;
+
+            //appIdParam = new SqlParameter("@APPLICATION_ID", apartAppId);
+
+            //// Get the hall choices that are already stored in the database for this application
+            //apartmentChoicesResult = RawSqlQuery<ApartmentChoiceSaveViewModel>.query("GET_AA_APARTMENT_CHOICES_BY_APP_ID @APPLICATION_ID", appIdParam); // run stored procedure
+            //if (apartmentChoicesResult == null)
+            //{
+            //    throw new ResourceNotFoundException() { ExceptionMessage = "The apartment preference could not be found." };
+            //}
+
+            //ApartmentChoiceSaveViewModel[] oldApartmentChoices = null;
+            //// Check whether any applicants were found matching the given application ID number
+            //if (apartmentChoicesResult.Any())
+            //{
+            //    oldApartmentChoices = new ApartmentChoiceSaveViewModel[apartmentChoicesResult.Count()];
+            //    for (int i = 0; i < apartmentChoicesResult.Count(); i++)
+            //    {
+            //        ApartmentChoiceSaveViewModel apartmentChoicesModel = apartmentChoicesResult.ElementAt(i);
+            //        oldApartmentChoices[i] = apartmentChoicesModel;
+            //    }
+            //}
+            //else
+            //{
+            //    oldApartmentChoices = new ApartmentChoiceSaveViewModel[0];
+            //}
+
+            //// Get an array of applicants IDs that are in the array recieved from the frontend but not yet in the database
+            //IEnumerable<string> applicantIdsToAdd = newApplicantIds.Except(oldApplicantIds);
```

Comment:
What's going on here? I see it is not ready by the variable names, so is it commented out for debugging those CORS errors and other unrelated errors that you were dealing this?

Reply:
This portion will be moved to a new branch apartapp-edit-hall. I'm working on it and made some changes so that I could work with viewmodel objects. They don't have '.Except'.

Reply:
The '.Except' is a method of the IEnumerable, not the ViewModel.
Though I can see how it might be a problem on custom objects, so you have to do the comparison based on the hall names only.

But that there should be existing code from the part I wrote which performs that same operation on the application but does all 3 comparisons in one for-loop, thus reducing the total number of times that we have to iterate through the array/list of applicants.

You could easily use a similar loop to compare the hall info, and it would make it easy to check for entries that have the same hall name but have a different ranking number (which means they need to be updated in the database rather than the insert or remove)

Reply:
Ok I will look for it. I was trying to use hashset just as shown in this [article](https://stackoverflow.com/questions/9998549/linq-use-except-on-collections-of-different-types-by-making-them-convertible/24436401).

Reply:
That was how I did it originally, but we changed it because we need to do several different comparisons between the same pair of arrays/lists/sets, so it is best to do all 3 comparisons in one iteration of the array, expect because the comparisons are not entirely independent so they logically overlap, making it even more efficient to do one loop (because it can actually be done using only a single `if` statement to construct the 3 possible results that we need, all in one `for` loop)